slidingwindow.h

```diff
@@ -1838,73 +1839,60 @@ void ConvolutionInputGenerator_1D_dilated_dws(
  */
-template<unsigned int ConvKernelDim_x,
-		 unsigned int IFMChannels,
-		 unsigned int Input_precision,
-		 unsigned int IFMDim_x,
-		 unsigned int OFMDim_x,
-		 unsigned int SIMD,
-		 typename R>
+template<
+	unsigned  ConvKernelDim_x,
+	unsigned  IFMChannels,
+	unsigned  Input_precision,
+	unsigned  IFMDim_x,
+	unsigned  OFMDim_x,
+	unsigned  Stride_x,
+	unsigned  SIMD,
+	typename  R		// Memory resource selector
+>
 void ConvolutionInputGenerator_1D_lowbuffer(
-		stream<ap_uint<SIMD*Input_precision> > & in,
-		stream<ap_uint<SIMD*Input_precision> > & out,
-		const unsigned int numReps,
-		R const &r) {
-	CASSERT_DATAFLOW(IFMChannels % SIMD == 0);
-	const unsigned int multiplying_factor = IFMChannels/SIMD;
-	const unsigned int buffer_size = ConvKernelDim_x * multiplying_factor;
-	ap_uint<SIMD*Input_precision> inputBuf[buffer_size];
-#pragma HLS ARRAY_PARTITION variable=inputBuf complete
-	memory_resource(inputBuf, r);
-	const unsigned int cycles_read_block = multiplying_factor*(ConvKernelDim_x-1)-(ConvKernelDim_x-1);
-	const unsigned int baseIter = cycles_read_block + (OFMDim_x * buffer_size);
-	const unsigned int num_of_inputs = IFMDim_x * multiplying_factor;
-	unsigned int inp = 0, index_write=0, index_read = 0, j = 0, internal_counter = 0;
-	for (unsigned int count_image = 0; count_image < numReps; count_image++) {
-		for (unsigned int i = 0; i < baseIter; i++) {
-			#pragma HLS PIPELINE II=1
-			if (inp < cycles_read_block) {// Initial buffer of ConvKernelDim lines
-				ap_uint<SIMD*Input_precision> inElem;
-				inElem = in.read();
-				inputBuf[inp] = inElem;
-				inp++;
-				index_write++;
-			}
-			else { // Read & write & update
-				// Read & write input buffer
-				if (inp < num_of_inputs){
-					if (inp < buffer_size || internal_counter >= buffer_size-multiplying_factor){
-						ap_uint<SIMD*Input_precision> inElem;
-						inElem = in.read();
-						index_write = index_write < buffer_size ? index_write : index_write - buffer_size;
-						inputBuf[index_write] = inElem;
-						inp++;
-						index_write++;
-						internal_counter = internal_counter < buffer_size-1 ? internal_counter+1 : 0;
-					}
-					else{
-						internal_counter++;
-					}
-				}
+	hls::stream<ap_uint<SIMD*Input_precision>> &in,
+	hls::stream<ap_uint<SIMD*Input_precision>> &out,
+	unsigned const  numReps,
+	R const &r
+) {
+	static_assert((IFMChannels % SIMD) == 0, "SIMD parallelism must divide the number of IFM channels");
+	static_assert(OFMDim_x == ((IFMDim_x-ConvKernelDim_x)/Stride_x+1), "Unexpected OFM dimension");
 
-				// Write output
-				ap_uint<SIMD*Input_precision> outElem = inputBuf[index_read];
-				out.write(outElem);
+	constexpr unsigned  SIMD_COUNT  = IFMChannels / SIMD;
+	constexpr unsigned  BUFFER_SIZE = (ConvKernelDim_x-1) * SIMD_COUNT;
+	constexpr unsigned  OUTPUT_SIZE = OFMDim_x * ConvKernelDim_x * SIMD_COUNT;
+	constexpr unsigned  NUM_OF_INPUTS = IFMDim_x*SIMD_COUNT;
```

Comment:
For consistency, `INPUT_SIZE` would be preferable.

Reply:
Thank you for the feedback, I've changed it.